<launch>
    <node pkg="pointcloud_to_laserscan" name="kinect_to_scan" type="pointcloud_to_laserscan_node">
        <remap from="cloud_in" to="/camera/depth/points"/>
        <remap from="scan" to="/laser/scan"/>

        <rosparam>
            target_frame: base_link # Leave disabled to output scan in pointcloud frame
            transform_tolerance: 0.01
            min_height: 0.2
            max_height: 3.0

            angle_min: -1.5708 # -M_PI/2
            angle_max: 1.5708 # M_PI/2
            angle_increment: 0.0157 # M_PI/200.0
            scan_time: 0.3333
            range_min: 0.45
            range_max: 5.0
            use_inf: true

            # Concurrency level, affects number of pointclouds queued for processing and number of threads used
            # 0 : Detect number of cores
            # 1 : Single threaded
            # 2->inf : Parallelism level
            concurrency_level: 1
        </rosparam>
    </node>
</launch>
